osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -339,6 +339,8 @@ public void DeleteDifficultyImmediately(BeatmapInfo beatmapInfo)
 
                 DeleteFile(setInfo, beatmapInfo.File);
                 setInfo.Beatmaps.Remove(beatmapInfo);
+                r.Remove(beatmapInfo.Metadata);
+                r.Remove(beatmapInfo);
```

Comment:
Notably, this same thing already shows up in a few places:

https://github.com/ppy/osu/blob/eb82bd38711843ffb6059e83e26ad1fa71e768a8/osu.Game/Beatmaps/BeatmapImporter.cs#L103-L104
https://github.com/ppy/osu/blob/eb82bd38711843ffb6059e83e26ad1fa71e768a8/osu.Game/Database/RealmAccess.cs#L347-L348

but I'm not super sure two lines are worth splitting a helper for. Can do if that's considered better.